src/Api/SecretsManager/Controllers/ServiceAccountsController.cs

```diff
@@ -105,6 +117,20 @@ public class ServiceAccountsController : Controller
             throw new NotFoundException();
         }
 
+        var org = await _organizationRepository.GetByIdAsync(organizationId);
+        if (org == null)
+        {
+            throw new NotFoundException();
+        }
```

Comment:
This is checked inside the query and command. I suggest moving this within the `if` statement so that you can avoid the call if it's not required (given that it's only required for the command), and drop the null check.

Reply:
Resolved in https://github.com/bitwarden/server/pull/3093/commits/0f3beda47fb78eddc0a67108e3f6922345c93dc5